### PR TITLE
networkd: do not change forwarding if IPForward is not set

### PR DESCRIPTION
**Pull request by mischief**

### CODE REVIEW

File: src/network/networkd.h [hunk 1]
```diff
@@ -60,6 +60,7 @@ typedef enum AddressFamilyBoolean {
         ADDRESS_FAMILY_IPV4 = 1,
         ADDRESS_FAMILY_IPV6 = 2,
         ADDRESS_FAMILY_YES = 3,
+        ADDRESS_FAMILY_UNSET = 4,
```
Comment (crawford) [inline src/network/networkd.h]: It's a little weird that "unset" isn't 0. I'm not sure if that change would break this though.

Comment (mischief) [inline src/network/networkd.h]: other parts of the code use the ADDRESS_FAMILY_* enums. changing the zero value would affect not only the ipforwarding code but also the other code using these enum values.

Comment (crawford) [inline src/network/networkd.h]: Right, and I'm sure somewhere someone is comparing the enum to an integer. :(

Comment (mischief) [inline src/network/networkd.h]: `grep -r ADDRESS_FAMILY_NO src/network/` in the systemd tree

**Comment by marineam** [comment]
LGTM, I will be curious if upstream will have an opinion on `0` meaning `NO` or `UNSET` but until then this is fine as-is.

**Comment by crawford** [comment]
+1

File: src/network/networkd.h (hunk 1 above)
Comment (marineam) [inline src/network/networkd.h]: @crawford actually, well, not comparing as an integer but just assuming that `NO` is `0` or boolean false. I'll bet all the other enum values could be ordered randomly. Damn 0 being so special ;-)

**Pull request merged**